api/src/test/java/org/openmrs/api/CohortServiceTest.java

```diff
@@ -644,5 +654,195 @@ public void getMemberships_shouldGetMembershipsAsOfADate() throws Exception {
+		newMember.setStartDate(startDate);
+		service.addMembershipToCohort(cohort, newMember);
+		
+		Date dateToTest = dateFormat.parse("2016-11-01 00:00:00");
+		List<CohortMembership> membersAsOfDate = cohort.getMemberships(dateToTest);
+		assertTrue(membersAsOfDate.isEmpty());
+		assertFalse(membersAsOfDate.stream().anyMatch(m -> m.getStartDate().equals(dateToTest)));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#isMembershipActive(Date)}
+	 */
+	@Test
+	@Verifies(value = "membership should be active as of a date", method = "isMembershipActive(Date)")
+	public void isMembershipActive_shouldBeActiveAsOfDate() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		newMember.setStartDate(startDate);
+		
+		Date dateToTest = dateFormat.parse("2017-01-01 12:00:00");
+		assertTrue(newMember.isMembershipActive(dateToTest));
+		assertTrue(newMember.isMembershipActive(new Date()));
+		assertTrue(newMember.isMembershipActive(startDate));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#isMembershipActive(Date)}
+	 */
+	@Test
+	@Verifies(value = "membership should be inactive as of a date", method = "isMembershipActive(Date)")
+	public void isMembershipActive_shouldBeInactiveAsOfDate() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		newMember.setStartDate(startDate);
+		
+		Date dateToTest = dateFormat.parse("2016-12-01 00:00:00");
+		assertFalse(newMember.isMembershipActive(dateToTest));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#isMembershipActive(Date)}
+	 */
+	@Test
+	@Verifies(value = "voided membership should be inactive as of a date", method = "isMembershipActive(Date)")
+	public void isMembershipActive_shouldBeInactiveAsOfDateForVoidedMembership() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		Date endDate = dateFormat.parse("2017-01-31 00:00:00");
+		newMember.setStartDate(startDate);
+		newMember.setEndDate(endDate);
+		newMember.setVoided(true);
+		
+		Date dateToTest = dateFormat.parse("2017-02-01 00:00:00");
+		assertFalse(newMember.isMembershipActive(dateToTest));
+		Date secondDateToTest = dateFormat.parse("2017-01-15 00:00:00");
+		assertFalse(newMember.isMembershipActive(secondDateToTest));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#compareTo(CohortMembership)}}
+	 */
+	@Test
+	@Verifies(value = "should compare two membership objects", method = "compareTo(CohortMembership)")
+	public void compareTo_shouldCompareMemberships() throws Exception {
```

Comment:
compareTo(). I wasn't entirely clear on how to perform this test. Should I be testing for equals() instead?